server/database/dbSeed.js

```diff
@@ -35,29 +35,73 @@ const databaseInsertion = (data) => new Promise((resolve, reject) => {
   });
 });
 
-const reviewGenerator = () => mocker()
-  .schema('reviewsTemplate', reviewsTemplate, Math.floor(Math.random() * 100))
+const reviewGenerator = (moonId, singleMoonEntry) => mocker()
+  .schema('reviewsTemplate', reviewsTemplate, singleMoonEntry || Math.floor(Math.random() * 100))
   .build()
   .then((info) => {
     const insertionPromises = [];
     info.reviewsTemplate.forEach((review) => {
-      const databaseData = [review.url_id, review.name, review.location, review.date,
-        review.comment, review.rating];
-      insertionPromises.push(databaseInsertion(databaseData));
+      const databaseData = [moonId || review.url_id,
+        review.name, review.location, review.date, review.comment, review.rating];
+
+      insertionPromises.push(databaseRawDataInserion(databaseData));
     });
     return Promise.all(insertionPromises)
       .catch((err) => console.error(err));
   });
 
-const databaseSeeder = () => {
-  let counter = (Math.floor(Math.random() * 15) + 5);
-  const databaseEntries = [];
-  while (counter > 0) {
-    databaseEntries.push(reviewGenerator());
-    counter -= 1;
+const databaseImageInsertion = (moonIndex, moonImageIndex) => new Promise((resolve, reject) => {
+  const randomIdIndex = Math.floor(Math.random() * 99) + 1; // 1 - 99
+  const randomImgIndex = Math.floor(Math.random() * 50) + 5; // 5 - 50
+  const imageURL = `https://hrr47-reviews.s3-us-west-1.amazonaws.com/${moonImageIndex || randomImgIndex}.jpg`;
+  const queryString = 'update user_reviews set img = "?" where url_id = ? and img is NULL order by rand() limit 1';
+  db.query(queryString, [imageURL, moonIndex || randomIdIndex],
+    (err, success) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(success);
+      }
+    });
+});
+
+const promiseCompiler = (counter, func, arg1, arg2) => {
+  const promiseArray = [];
+  let loopCounter = counter;
+  while (loopCounter > 0) {
+    promiseArray.push(func(arg1, arg2));
+    loopCounter -= 1;
   }
-  return Promise.all(databaseEntries)
-    .then(() => console.log('🚀🚀 reviews database seeded!'))
+  return promiseArray;
+};
+
+const databaseSeeder = () => {
+  const mainReviewCounter = Math.floor(Math.random() * 25) + 10;
+  return Promise.all(promiseCompiler(mainReviewCounter, reviewGenerator))
+    .then(() => console.log('🚀🚀 review database seeded!'))
+    .then(() => {
+      const imageCounter = Math.floor(Math.random() * 300) + 100;
+      return Promise.all(promiseCompiler(imageCounter, databaseImageInsertion))
```

Comment:
@Matt-Wigg , you are calling the promiseCompiler with two arguments. I see that it takes four. Is this working?

Reply:
I can see that the parameters in the referenced functions are optional. This should be ok then.